FlowCrypt/src/test/java/com/flowcrypt/email/security/pgp/PgpKeyTest.kt

```diff
@@ -126,4 +129,16 @@ class PgpKeyTest {
     assertEquals(1, result.getAllKeys().size)
     assertEquals(expected, result.toPgpKeyDetailsList().first())
   }
+
+  @Test
+  fun testParseAndDecryptKey_Issue1296() {
+    val publicKeyRing = Utils.loadPublicKey("issue-1296-0xA96B4C55A800DB83.public.gpg-key")
+    val fingerprint = publicKeyRing.publicKey.fingerprint.toHexString()
+    val secretKeyRing = Utils.loadSecretKey("issue-1296-0xA96B4C55A800DB83.secret-subkeys.gpg-key")
+    assertEquals(fingerprint, secretKeyRing.publicKey.fingerprint.toHexString())
+    val passphrase = Passphrase.fromPassword("password12345678")
+    val decryptedSecretKeyRing = PgpKey.decryptKey(secretKeyRing, passphrase)
+    val decryptedKeyFingerprint = decryptedSecretKeyRing.publicKey.fingerprint.toHexString()
+    assertEquals(fingerprint, decryptedKeyFingerprint)
+  }
```

Comment:
Maybe it would be better to use PGPainless for the fingerprint comparison

```
@Test
  fun testParseAndDecryptKey_Issue1296() {
    val publicKeyRing = Utils.loadPublicKey("issue-1296-0xA96B4C55A800DB83.public.gpg-key")
    val expectedFingerprint = OpenPgpV4Fingerprint(publicKeyRing)
    val secretKeyRing = Utils.loadSecretKey("issue-1296-0xA96B4C55A800DB83.secret-subkeys.gpg-key")
    assertEquals(expectedFingerprint, OpenPgpV4Fingerprint(secretKeyRing.publicKey))
    val passphrase = Passphrase.fromPassword("password12345678")
    val decryptedSecretKeyRing = PgpKey.decryptKey(secretKeyRing, passphrase)
    assertEquals(expectedFingerprint, OpenPgpV4Fingerprint(decryptedSecretKeyRing.publicKey))
  }
```

Reply:
In that case, you don't need `ByteArray.toHexString`

Reply:
Hmm, maybe, will try.